### PR TITLE
Problem: gas_used differs in getting of consensusParams when enable cache

### PR DESCRIPTION
**Pull request by mmsqe**
Revert "Problem: signature verification result not cache between incarnations of same tx (#565)"

This reverts commit 5a1594f17924388dc2530bd8143ca2b04655fad4.

# Description

Closes: #XXXX

---

## Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] provided a link to the relevant issue or specification
* [ ] reviewed "Files changed" and left comments if necessary
* [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
* [ ] added a changelog entry to `CHANGELOG.md`
* [ ] updated the relevant documentation or specification, including comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] confirmed all CI checks have passed

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic, API design and naming, documentation is accurate, tests and test coverage

### CODE REVIEW

**Review by yihuang — COMMENTED** [review]
Can you describe the issue first, and do we run out of options to fix it?

EDIT: I think it's fine to revert cosmos-sdk part, since the ethermint side is used most.

**Review by yihuang — APPROVED** [review]

**Comment by mmsqe** [comment]
I'm not sure if we can align the gas_used since there'd be less GetConsensusParams when enable cache. Or do you mean we record the gas_used like https://github.com/crypto-org-chain/cosmos-sdk/commit/f33944ee849c66f1b51d5d29e6e4f3b6ce41797a

File: baseapp/abci.go [hunk 1]
```diff
@@ -842,8 +842,8 @@ func (app *BaseApp) internalFinalizeBlock(ctx context.Context, req *abci.Request
 
 func (app *BaseApp) executeTxs(ctx context.Context, txs [][]byte) ([]*abci.ExecTxResult, error) {
 	if app.txExecutor != nil {
-		return app.txExecutor(ctx, len(txs), app.finalizeBlockState.ms, func(i int, ms storetypes.MultiStore, incarnationCache map[string]any) *abci.ExecTxResult {
-			return app.deliverTxWithMultiStore(txs[i], i, ms, incarnationCache)
+		return app.txExecutor(ctx, len(txs), app.finalizeBlockState.ms, func(i int, ms storetypes.MultiStore) *abci.ExecTxResult {
```
Comment (yihuang) [inline baseapp/abci.go:845]: txExecutor interface need to keep too.

Comment (mmsqe) [reply baseapp/abci.go:845]: we just [need](https://github.com/crypto-org-chain/ethermint/blob/0e0baa4ea54079296cebf233fe7510700f981830/app/ante/handler_options.go#L116-L122) get and set right? Do you mean keep txExecutor but pass nil?

Comment (yihuang) [reply baseapp/abci.go:845]: [stm executor](https://github.com/crypto-org-chain/ethermint/blob/develop/app/executor.go#L66) will pass non-nil cache.

**Comment by mmsqe** [comment]
Diff doesn't come from consensusParams but from auth ante handler which was fixed [before](https://github.com/crypto-org-chain/cosmos-sdk/pull/570/files)

**Pull request closed without merging**